src/amo/components/ReportAbuseButton/index.js

```diff
@@ -207,7 +212,7 @@ export class ReportAbuseButtonBase extends React.Component {
   }
 }
 
-export const mapStateToProps = (state, ownProps) => {
+export const mapStateToProps = (state: Object, ownProps: Props) => {
```

Comment:
We already have this state type (although it could use a better name) -- we might as well use it. I don't see any errors with:

```diff
diff --git a/src/amo/components/ReportAbuseButton/index.js b/src/amo/components/ReportAbuseButton/index.js
index aeb9292b..31f4dbfb 100644
--- a/src/amo/components/ReportAbuseButton/index.js
+++ b/src/amo/components/ReportAbuseButton/index.js
@@ -19,6 +19,7 @@ import {
 } from 'core/reducers/abuse';
 import { sanitizeHTML } from 'core/utils';
 import Button from 'ui/components/Button';
+import type { ReducerState } from 'core/reducers/abuse';
 import type { AddonType } from 'core/types/addons';
 import type { I18nType } from 'core/types/i18n';
 
@@ -212,7 +213,9 @@ export class ReportAbuseButtonBase extends React.Component<Props> {
   }
 }
 
-export const mapStateToProps = (state: Object, ownProps: Props) => {
+export const mapStateToProps = (
+  state: {| abuse: ReducerState |}, ownProps: Props
+) => {
   const addon = ownProps.addon;
 
   return {
diff --git a/src/core/reducers/abuse.js b/src/core/reducers/abuse.js
index 6dd2c7d7..38bc84a6 100644
--- a/src/core/reducers/abuse.js
+++ b/src/core/reducers/abuse.js
@@ -127,7 +127,7 @@ export const initialState = {
   loading: false,
 };
 
-type ReducerState = {|
+export type ReducerState = {|
   bySlug: {
     [addonSlug: string]: {|
       buttonEnabled?: bool,
```